### PR TITLE
Feature/cereal example

### PR DESCRIPTION
**Pull request by DeveloperPaul123**
This adds a further example for the [`cereal`](https://github.com/USCiLab/cereal) library. 

Note that this builds on top of PR #84 and merging this will merge that PR as well. I separated them out because this example had build warnings and wasn't sure if that is going to be a problem.

### CODE REVIEW

File: examples/cereal/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# ---- Create binary ----
+
+add_executable(CPMExampleCereal main.cpp)
+target_link_libraries(CPMExampleCereal cereal )
+set_target_properties(CPMExampleCereal PROPERTIES CXX_STANDARD 17)
+
+# ---- Add code coverage ----
+
+if (${ENABLE_TEST_COVERAGE})
```
Comment (TheLartians) [inline examples/cereal/CMakeLists.txt]: For the sake of minimal examples I would restrict adding test coverage to testing-related examples. The coverage is only there as an illustration for a complete test and not actually used by the CI or anything in this project.

File: examples/cereal/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
```
Comment (TheLartians) [inline examples/cereal/CMakeLists.txt]: As in #84:
```suggestion

project(CPMExampleCereal)

```

File: examples/cereal/CMakeLists.txt (hunk 1 above)
Comment (DeveloperPaul123) [reply examples/cereal/CMakeLists.txt]: Ok sounds good, I wasn't 100% sure about that.

**Comment by DeveloperPaul123** [comment]
I've update the wiki.

File: examples/cereal/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+project(CPMExampleCereal)
+
+# ---- Options ----
+
+option(ENABLE_TEST_COVERAGE "Enable test coverage" OFF)
```
Comment (TheLartians) [inline examples/cereal/CMakeLists.txt]: Just realised that this option can be removed now as well.

**Pull request merged**